packages/framework/src/Model/Article/ArticleFriendlyUrlDataProvider.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Shopsys\FrameworkBundle\Model\Article;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Query\Expr\Join;
+use Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig;
+use Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\CompilerPass\FriendlyUrlDataProviderInterface;
+use Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\FriendlyUrl;
+use Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\FriendlyUrlData;
+
+class ArticleFriendlyUrlDataProvider implements FriendlyUrlDataProviderInterface
+{
+    private const ROUTE_NAME = 'front_article_detail';
+    /**
+     * @var \Doctrine\ORM\EntityManagerInterface
+     */
+    private $em;
+
+    public function __construct(
+        EntityManagerInterface $em
+    ) {
+        $this->em = $em;
+    }
+
+    /**
+     * Returns data required for friendly url generator
```

Comment:
I believe it is not necessary to describe the method here (and all the other implementations) as it is described in the interface itself :wink:

Reply:
the same applies to `getRouteName()` method :slightly_smiling_face: